### PR TITLE
adding module azure_rm_virtualmachinescalesetinstance_facts

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
This module provides functionality to get facts about VM instances in VMSS.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
azure_rm_virtualmachinescalesetinstance_facts

### CODE REVIEW

**Comment by zikalino** [comment]
will add test in a moment

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetvm_facts.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_virtualmachinescalesetvm_facts
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetvm_facts.py]: how about virtualmachinescalsetinstance_facts module? to align term with both rest and sdk

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetvm_facts.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            returned: always
+            type: str
+            sample: 0
+        name:
+            description:
+                - Virtual Machine name
+            returned: always
+            type: str
+            sample: myVMSS_2
+        latest_model_applied:
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetvm_facts.py]: the returned info is too limited, what about vnet/subnet, disk, sku, status(power_state), which showed at portal?

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetvm_facts.py (hunk 1 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetvm_facts.py]: yes, I was thinking about exactly that change.. I will change it

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetvm_facts.py (hunk 2 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetvm_facts.py]: there can be quite a lot of info here... that's why i wanted to limit returned values in the first release.

Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetvm_facts.py]: but i think i can add some more ;-)

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetinstance_facts.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+EXAMPLES = '''
+  - name: List VM instances in Virtual Machine ScaleSet
+    azure_rm_computevirtualmachinescalesetinstance_facts:
+      resource_group: myResourceGroup
+      vmss_name: myVMSS
+'''
+
+RETURN = '''
+instances:
+    description: A list of dictionaries containing facts for Virtual Machine Scale Set V M.
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetinstance_facts.py]: extra whitespace between V and M.

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetinstance_facts.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+            type: bool
+            sample: True
+        provisioning_state:
+            description:
+                - Provisioning state of the Virtual Machine
+            returned: always
+            type: str
+            sample: Succeeded
+        vm_id:
+            description:
```
Comment (yungezz) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetinstance_facts.py:105]: could you pls add status(powerstate) in facts?

**Review by yungezz — COMMENTED** [review]
LGTM. need more returned properties.

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetinstance_facts.py (hunk 4 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetinstance_facts.py:105]: added

File: lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetinstance_facts.py (hunk 3 above)
Comment (zikalino) [reply lib/ansible/modules/cloud/azure/azure_rm_virtualmachinescalesetinstance_facts.py]: fixed

**Review by yungezz — APPROVED** [review]

**Comment by zikalino** [comment]
shipit

**Comment by yungezz** [comment]
shipit

**Pull request merged**